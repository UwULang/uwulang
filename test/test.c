/* 
* UwULang (MIT)
*/ 
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <time.h>

struct Block {
    char value;
    struct Block* next;
    struct Block* parent;
} UwU = {0, NULL, NULL};

struct Block* next(struct Block* curr) {
    if (curr->next != NULL) {
        return curr->next;
    } else {
        curr->next = malloc(sizeof (struct Block));
        curr->next->value = 0;
        // if next cannot be alloc
        if (curr->next == NULL)
            return NULL;
        else 
            return curr->next;
    }
}

/// Main takes in either a single file with file extension uwu
/// or if no args are present will run interperted on the command line
int main(int argc, char *argv[]) {
    struct Block* head = &UwU;
    struct Block* curr = &UwU;
    // Use current time as seed for random generator
    srand(time(0));

    // TODO: convert char to UTF-8 emojis

    // inital way
    char code[32768] = {0};
    int *inputCounter = code;
    int currInputChar;

    // testing emojis
redo:
    while ((currInputChar = getchar()) != EOF && currInputChar != '\n') {
        printf("%d ", currInputChar);
    }
    goto redo;

    // since struct is statically defined, no need to dealloc

    // new line
    putchar('\n');


    return 0;
}

// tests
// >>++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++. ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ‘†ğŸ¥º
