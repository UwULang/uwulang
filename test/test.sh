#!/bin/bash
echo "Testing output:"
./uwulang test/hellOwOrld.uwu > test1.out
./uwulang test/sqwuare10000.uwu > test2.out

diff test1.out test/hello.out > diff.out
diff test2.out test/square.out >> diff.out

if [ -s diff.out ] # test if diff is empty
then
    cat diff.out # print
    exit 1 # test failed
fi