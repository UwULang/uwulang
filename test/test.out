👆
240 159 145 134 
👇
240 159 145 135 
👉
240 159 145 137 
👈
240 159 145 136 
🥺
240 159 165 186 
😳
240 159 152 179 
🥴
240 159 165 180 
😒
240 159 152 146 
😡
240 159 152 161